### ISSUE TITLE
Possible Memory Leak

### ISSUE THREAD

**Issue by lczvd** [stephencelis/SQLite.swift#1162]
Hi, 

I just noticed something on version 0.13.3. 

The Xcode Instruments is reporting the following leak:

![leak](https://user-images.githubusercontent.com/115730747/195660063-f8ffb2ee-6544-44e0-8e59-fdb1fcc92f77.png)

By the way, I'm using Xcode 14.0.1 and the app is running on iOS 14.6.

Any ideas on the possible source of this issue?

Thanks in advance!

**Comment by NathanFallet** [comment]
More information is required. Can you provide a reproducible example?

**Comment by davifmarcelino** [comment]
Hi,
I have the same issue so I will give you guys the stack trace of 3 sqlcipher_mem_malloc leak in the project
<img width="847" alt="AllLeaks" src="https://user-images.githubusercontent.com/48737782/213513007-aa38c530-a24e-45a3-847f-080d3d7e3424.png">

the first leak
<img width="782" alt="Leak_1" src="https://user-images.githubusercontent.com/48737782/213513195-b269855d-d034-4bb1-b744-1f98be61bdd5.png">

the second leak
<img width="792" alt="Leak_2-1" src="https://user-images.githubusercontent.com/48737782/213513252-b81596ee-a9d1-4037-8ec7-b8e3fb1e9b29.png">
<img width="787" alt="Leak_2-2" src="https://user-images.githubusercontent.com/48737782/213513257-b65b7629-84c4-4c16-bc69-dca43d7295b5.png">

and the last leak
<img width="786" alt="Leak_3-1" src="https://user-images.githubusercontent.com/48737782/213513582-19201852-8d61-4fe7-a9b9-946e974cf570.png">
<img width="784" alt="Leak_3-2" src="https://user-images.githubusercontent.com/48737782/213513588-5eff720f-11d1-40ac-8d30-1d95ea78aff6.png">

**Comment by raulperesgoes** [comment]
Do you need more info to work on that? Is there any other way we could help you?

**Comment by GuillaumeUnice** [comment]
I encounter the exact same problem... any improvement?

**Comment by sinhpn92** [comment]
Did you resolve this issue @raulperesgoes ? @GuillaumeUnice

**Comment by GuillaumeUnice** [comment]
Yes on my side it was due to the fact I open the database again at each SQL query and behind the hood is using a custom malloc in C and I never destroy the ref on it as I don't close the previous open DB.

Basically, I wrongly use the lib
